conans/test/integration/workspace_test.py

```diff
@@ -80,166 +64,255 @@ def package_info(self):
         client.save(files("B", "C"), path=os.path.join(client.current_folder, "B"))
+        A = os.path.join(client.current_folder, "A")
         a = files("A", "B")
-        a["src/CMakeLists.txt"] += "add_executable(app main.cpp)\ntarget_link_libraries(app helloA)\n"
+        a["src/CMakeLists.txt"] += ("add_executable(app main.cpp)\n"
+                                    "target_link_libraries(app helloA)\n")
         a["src/main.cpp"] = main_cpp
-        client.save(a, path=os.path.join(client.current_folder, "A"))
-
-        project = """HelloB:
-    folder: B
-    includedirs: src
-    cmakedir: src
-HelloC:
-    folder: C
-    includedirs: src
-    cmakedir: src
-HelloA:
-    folder: A
-    cmakedir: src
-
-root: HelloA
-generator: cmake
-name: MyProject
-"""
-        client.save({WORKSPACE_FILE: project})
-        client.run("install . -if=build")
+        client.save(a, path=A)
+
+        project = dedent("""
+            HelloB/0.1@lasote/stable:
+                folder: B
+            HelloC/0.1@lasote/stable:
+                folder: C
+            HelloA/0.1@lasote/stable:
+                folder: A
+            layout: layout
+            root: HelloA/0.1@lasote/stable
+            """)
+        layout = dedent("""
+            [folders]
+            build: build/{settings.build_type}
+
+            [includedirs]
+            src
+
+            [libdirs]
+            build/{settings.build_type}/lib
+            """)
+        client.save({"conanws.yml": project,
+                     "layout": layout})
+        client.run("workspace install conanws.yml")
+        client.run("workspace install conanws.yml -s build_type=Debug")
+        self.assertIn("HelloA/0.1@lasote/stable from user folder - Editable", client.out)
+        self.assertIn("HelloB/0.1@lasote/stable from user folder - Editable", client.out)
+        self.assertIn("HelloC/0.1@lasote/stable from user folder - Editable", client.out)
+
+        build_type = "Release"
+        client.run("build C -bf=C/build/%s" % build_type)
+        client.run("build B -bf=B/build/%s" % build_type)
+        client.run("build A -bf=A/build/%s" % build_type)
+
+        cmd_release = os.path.normpath("./A/build/Release/bin/app")
+        cmd_debug = os.path.normpath("./A/build/Debug/bin/app")
+
+        client.runner(cmd_release, cwd=client.current_folder)
+        self.assertIn("Hello World C Release!", client.out)
+        self.assertIn("Hello World B Release!", client.out)
+        self.assertIn("Hello World A Release!", client.out)
+
+        build_type = "Debug"
+        client.run("build C -bf=C/build/%s" % build_type)
+        client.run("build B -bf=B/build/%s" % build_type)
+        client.run("build A -bf=A/build/%s" % build_type)
+
+        client.runner(cmd_debug, cwd=client.current_folder)
+        self.assertIn("Hello World C Debug!", client.out)
+        self.assertIn("Hello World B Debug!", client.out)
+        self.assertIn("Hello World A Debug!", client.out)
```

Comment:
Parameterize this test for `build_type` in `Debug|Release`

Reply:
No, I don't want to do that. I want to run one build after the other. That is the key of this test, in fact that helped to solve many bugs. Behavior is really affected by dealing with both builds.